training/CHAPEL_TUTORIAL.md

```diff
@@ -993,358 +1073,635 @@ Expected Output:
     }
+    writeln();
+    writeln("src: ", src);
+    writeln("dst: ", dst);
 }
+```
 
-config param SIZE = 1000000;
-const distDom = Block.createDomain({0..#SIZE});
+using a `DstAggregator`:
+```console
+$ chpl tutorial.chpl --no-cache-remote
+$ ./tutorial -nl 2
 
-// create a array containing tuples of uints
-var hashes: [distDom] (uint, uint) = (1, 1):(uint, uint);
+src: 0 1 2 3 4 5
+dst: 0 0 0 0 0 0
+
+iteration num: 0
+  on Locale: 0
+  on localSubDom: {0..2}
+  src[0] is local
+  dst[3] is remote
+
+iteration num: 2
+  on Locale: 0
+  on localSubDom: {0..2}
+  src[2] is local
+  dst[5] is remote
+
+iteration num: 1
+  on Locale: 0
+  on localSubDom: {0..2}
+  src[1] is local
+  dst[4] is remote
+
+iteration num: 3
+  on Locale: 1
+  on localSubDom: {3..5}
+  src[3] is local
+  dst[0] is remote
+
+iteration num: 5
+  on Locale: 1
+  on localSubDom: {3..5}
+  src[5] is local
+  dst[2] is remote
+
+iteration num: 4
+  on Locale: 1
+  on localSubDom: {3..5}
+  src[4] is local
+  dst[1] is remote
+
+src: 0 1 2 3 4 5
+dst: 3 4 5 0 1 2
+```
+
+using a `SrcAggregator`:
+```console
+$ ./tutorial -nl 2 --UseDstAgg=false
+src: 0 1 2 3 4 5
+dst: 0 0 0 0 0 0
+
+iteration num: 0
+  on Locale: 0
+  on localSubDom: {0..2}
+  src[3] is remote
+  dst[0] is local
+
+iteration num: 1
+  on Locale: 0
+  on localSubDom: {0..2}
+  src[4] is remote
+  dst[1] is local
+
+iteration num: 2
+  on Locale: 0
+  on localSubDom: {0..2}
+  src[5] is remote
+  dst[2] is local
+
+iteration num: 3
+  on Locale: 1
+  on localSubDom: {3..5}
+  src[0] is remote
+  dst[3] is local
+
+iteration num: 5
+  on Locale: 1
+  on localSubDom: {3..5}
+  src[2] is remote
+  dst[5] is local
+
+iteration num: 4
+  on Locale: 1
+  on localSubDom: {3..5}
+  src[1] is remote
+  dst[4] is local
+
+src: 0 1 2 3 4 5
+dst: 3 4 5 0 1 2
+```
 
-var upperEntry = new SymEntry(SIZE, uint);
-var lowerEntry = new SymEntry(SIZE, uint);
+<a id="perf"></a>
+## Performance and Diagnostics
+Now that you're a pro at writing chapel code, let's talk about how to make the code you write more efficient!
 
-resetCommDiagnostics();
-startCommDiagnostics();
-var t1 = Time.timeSinceEpoch().totalSeconds();
+<a id="var_dec"></a>
+### Variable Declarations
+Throughout this tutorial we've mainly used `var` to declare variables, but this is not the only option.
+The driving factor of using other declarations is to improve performance or reduce memory usage.
 
-// the leading iterator is a range, so all the computation happens on locale 0
-forall (i, (up, low)) in zip(0..#SIZE, hashes) {
-  upperEntry.a[i] = up;
-  lowerEntry.a[i] = low;
-}
+[`const`](https://chapel-lang.org/docs/users-guide/base/constParam.html):
+* can be used when we know a variable should never be changed
+* a program execution time constant. This means the compiler doesn't need to know its value, and it can be 
+set at runtime (i.e. passed in by a user)
+* knowing this value won't change can help the compiler to make optimizations
```

Comment:
You could add something in this list about it being a way for a programmer to keep themselves honest / check themselves.  I.e., "I don't want this to ever change, and want the compiler to complain at me if I do try to re-assign it."

I also wonder whether the final bullet should be strengthened by saying something like "programmers should always reach for `const` when they have variables that they know won't be changing [for both the performance and code safety benefits]"